### ISSUE TITLE
Different disk size from disk.getinfo than  from "lsblk" Linux command

### ISSUE THREAD

**Issue by Dinaamagdy** [g8os/core0#134]
#### Results from disk.getinfo(disk_name)
![selection_246](https://cloud.githubusercontent.com/assets/24386802/24751095/969c419c-1ac8-11e7-9127-5ace6759516e.png)
#### Results from  lsblk 
![selection_247](https://cloud.githubusercontent.com/assets/24386802/24751185/e5c7fc34-1ac8-11e7-8a46-ebac0c9ad245.png)

**Comment by maxux** [comment]
Can't reproduce. Do you have more context ?

```
In [34]: cl.disk.getinfo('sda')
Out[34]: 
{'alignment': '0',
 'blocksize': 512,
 'children': None,
 'disc-aln': '0',
 'disc-gran': '512B',
 'disc-max': '2G',
 'disc-zero': '0',
 'end': 4294967295,
 'free': [],
 'fstype': None,
 'group': 'root',
 'hctl': '0:0:0:0',
 'hotplug': '0',
 'kname': 'sda',
 'label': None,
 'log-sec': '512',
 'maj:min': '8:0',
 'min-io': '512',
 'mode': 'brw-------',
 'model': 'QEMU HARDDISK   ',
 'mountpoint': None,
 'name': 'sda',
 'opt-io': '0',
 'owner': 'root',
 'partflags': None,
 'partlabel': None,
 'parttype': None,
 'partuuid': None,
 'phy-sec': '512',
 'pkname': None,
 'ra': '128',
 'rand': '1',
 'rev': '2.5+',
 'rm': '0',
 'ro': '0',
 'rota': '1',
 'rq-size': '128',
 'sched': 'cfq',
 'serial': '',
 'size': 4294967296,
 'start': 0,
 'state': 'running',
 'subsystems': 'block:scsi:pci',
 'table': 'loop',
 'tran': 'ata',
 'type': 'disk',
 'uuid': None,
 'vendor': 'ATA     ',
 'wsame': '0B',
 'wwn': None}

In [35]: cl.bash('lsblk -O -P -b /dev/sda').get()
Out[35]: 
STATE: SUCCESS
STDOUT:
NAME="sda" KNAME="sda" MAJ:MIN="8:0" FSTYPE="" MOUNTPOINT="" LABEL="" UUID="" PARTTYPE="" PARTLABEL="" PARTUUID="" PARTFLAGS="" RA="128" RO="0" RM="0" HOTPLUG="0" MODEL="QEMU HARDDISK   " SERIAL="" SIZE="4294967296" STATE="running" OWNER="root" GROUP="root" MODE="brw-------" ALIGNMENT="0" MIN-IO="512" OPT-IO="0" PHY-SEC="512" LOG-SEC="512" ROTA="1" SCHED="cfq" RQ-SIZE="128" TYPE="disk" DISC-ALN="0" DISC-GRAN="512" DISC-MAX="2147450880" DISC-ZERO="0" WSAME="0" WWN="" RAND="1" PKNAME="" HCTL="0:0:0:0" TRAN="ata" SUBSYSTEMS="block:scsi:pci" REV="2.5+" VENDOR="ATA     "

STDERR:

DATA:
```

```
 'size': 4294967296,
SIZE="4294967296"
```

**Comment by Dinaamagdy** [comment]
No , I did same steps ,and still I can reproduce it as shown 
```
In [125]: c.disk.getinfo('sda')
Out[125]: 
{'alignment': '0',
 'blocksize': 4096,
 'children': None,
 'disc-aln': '0',
 'disc-gran': '512B',
 'disc-max': '2G',
 'disc-zero': '1',
 'end': 1200319561727,
 'free': [],
 'fstype': None,
 'group': 'root',
 'hctl': '0:0:0:0',
 'hotplug': '1',
 'kname': 'sda',
 'label': None,
 'log-sec': '512',
 'maj:min': '8:0',
 'min-io': '4096',
 'mode': 'brw-------',
 'model': 'INTEL SSDSC2BB15',
 'mountpoint': '/var/cache',
 'name': 'sda',
 'opt-io': '0',
 'owner': 'root',
 'partflags': None,
 'partlabel': None,
 'parttype': None,
 'partuuid': None,
 'phy-sec': '4096',
 'pkname': None,
 'ra': '128',
 'rand': '0',
 'rev': '0101',
 'rm': '1',
 'ro': '0',
 'rota': '0',
 'rq-size': '128',
 'sched': 'noop',
 'serial': '',
 'size': 1200319561728,
 'start': 0,
 'state': 'running',
 'subsystems': 'block:scsi:pci',
 'table': 'loop',
 'tran': 'sata',
 'type': 'disk',
 'uuid': None,
 'vendor': 'ATA     ',
 'wsame': '0B',
 'wwn': None}
```
```
In [127]: c.bash(' lsblk -O -P -b -d dev/sda ').get()
Out[127]: 
STATE: SUCCESS
STDOUT:
NAME="sda" KNAME="sda" MAJ:MIN="8:0" FSTYPE="" MOUNTPOINT="/var/cache" LABEL="" UUID="" PARTTYPE="" PARTLABEL="" PARTUUID="" PARTFLAGS="" RA="128" RO="0" RM="1" HOTPLUG="1" MODEL="INTEL SSDSC2BB15" SERIAL="" SIZE="150039945216" STATE="running" OWNER="root" GROUP="root" MODE="brw-------" ALIGNMENT="0" MIN-IO="4096" OPT-IO="0" PHY-SEC="4096" LOG-SEC="512" ROTA="0" SCHED="noop" RQ-SIZE="128" TYPE="disk" DISC-ALN="0" DISC-GRAN="512" DISC-MAX="2147450880" DISC-ZERO="1" WSAME="0" WWN="" RAND="0" PKNAME="" HCTL="0:0:0:0" TRAN="sata" SUBSYSTEMS="block:scsi:pci" REV="0101" VENDOR="ATA     
```
You can connect on this machine main9 in packet.net with ip 147.75.102.239

**Comment by maxux** [comment]
@muhamadazmy Here is the fix:
https://github.com/g8os/core0/blob/1.1.0-alpha-issue-133/core0/builtin/disk.go#L154-L184

This block of code need to be replaced. This gives you the physical block size, this is not correct, the size reported by `/sys/block/..../size` is to multiply by logical block size, not physical.

You can get them (logical and physical) more easily by reading theses files than running `lsblk`:
- `/sys/block/sda/queue/physical_block_size`
- `/sys/block/sda/queue/logical_block_size`

**Comment by maxux** [comment]
According to @Dinaamagdy, fixed using `1.1.0-alpha-issue-134` branch.

**Comment by john-kheir** [comment]
## Verified on 1.1.0-alpha
